tabs/pid_tuning.js

```diff
@@ -701,11 +718,8 @@ TABS.pid_tuning.initRatesPreview = function () {
     this.keepRendering = true;
     this.model = new Model($('.rates_preview'), $('.rates_preview canvas'));
 
-    var scale = d3.scale.linear().domain([900, 2100]);
-
-    this.rollScale = scale.range([Math.PI * 2, -Math.PI * 2]);
-    this.pitchScale = scale.range([Math.PI * 2, -Math.PI * 2]);
-    this.yawScale = scale.range([Math.PI * 2, -Math.PI * 2]);
+    this.$superExpo = $('.pid_tuning input[name="show_superexpo_rates"]');
```

Comment:
Well without code reviews pretty much anything is a recipe for disaster. But `_`-private methods is a fairly known pattern in JS community, so I wouldn't expect a lot of problems with it.

I'm sorry, but I keep noticing that you sound a bit like "let's not change anything but stick to the current mess, because this is what people are used to". No offence here, pure subjective observation.

`this.foo.private.bar.private.baz` shouldn't be a thing because you're clearly calling a `private` method on something that you should have no access to (by convention it is), but I agree, don't like this approach myself.

Non-nested closures seems like a much better option than the one we have right now. Although I still think same approach (with a difference of prefixing private methods) here is better and less error-prone.

I'm used to `snake_case` for variables and `camelCase` for methods (which is the case in many JS projects), although I'm very flexible here if there's a common agreement. What case do you prefer? `currentRates.roll_rate` had to match the name of the input in order to do dynamic updates you see in:

```javascript
if (self.currentRates.hasOwnProperty(targetElement.attr('name')) && targetValue) {
  self.currentRates[targetElement.attr('name')] = targetValue * 100;
}
```

P.S. Re the tools — that's pretty much what I use for JS development. I'm sure there are some IDE's, but I don't think they give you much more in terms of debugging, especially when it's a Chrome-app.